### ISSUE TITLE
Cannot gsync with Python2 if gsync was used with Python3

### ISSUE THREAD

**Issue by molinav** [molinav/gsync#9]
Steps to reproduce the bug:
1.  Run `gsync` using Python 3:
```
python3 bin/gsync -r -v --progress testdata/ drive://testdata
```
2. Run the same command with `gsync` but using Python 2:
```
python2 bin/gsync -r -v --progress testdata/ drive://testdata
```
3.  Profit:
```
DEBUG: TypeError("'NoneType' object is not iterable",):
'NoneType' object is not iterable
DEBUG: TypeError("'NoneType' object is not iterable",):
  File "/home/vic/github/python-gsync/libgsync/crawler.py",
  line 194, in run
    self._walk(srcpath, self._walk_callback, self._dev)
  File "/home/vic/github/python-gsync/libgsync/crawler.py",
  line 154, in _walk
    self._sync(dirpath)
  File "/home/vic/github/python-gsync/libgsync/sync/__init__.py",
  line 262, in __call__
    self._sync(path)
  File "/home/vic/github/python-gsync/libgsync/sync/__init__.py",
  line 320, in _sync
    self.dst.update_attrs(dst_path, src_file)
  File "/home/vic/github/python-gsync/libgsync/sync/file/__init__.py",
  line 482, in update_attrs
    self.__update_attrs(path, src_obj)
  File "/home/vic/github/python-gsync/libgsync/sync/file/__init__.py",
  line 413, in __update_attrs
    self._update_attrs(path, src, attrs)
  File "/home/vic/github/python-gsync/libgsync/sync/file/remote.py",
  line 165, in _update_attrs
    st_info = list(tuple(info.statInfo))

Error: TypeError("'NoneType' object is not iterable",)
sent 0 bytes  received 0 bytes  0.00 bytes/sec
```

**Comment by molinav** [comment]
So the pull request #10 fixes one of the reasons why the code reaches this `TypeError`. I still get the same error, but I realised that some broad try-except clauses are really hiding the root of the problem (see [`libgsync/sync/file/__init__.py`](../tree/c6776348/libgsync/sync/file/__init__.py#L261)). If I add `raise` in the except-clause to force the error to be raised, we get more info:
```
DEBUG: ValueError('unsupported pickle protocol: 4',):
unsupported pickle protocol: 4
DEBUG: ValueError('unsupported pickle protocol: 4',):
  File "/home/vic/github/python-gsync/libgsync/crawler.py",
  line 194, in run
    self._walk(srcpath, self._walk_callback, self._dev)
  File "/home/vic/github/python-gsync/libgsync/crawler.py",
  line 154, in _walk
    self._sync(dirpath)
  File "/home/vic/github/python-gsync/libgsync/sync/__init__.py",
  line 262, in __call__
    self._sync(path)
  File "/home/vic/github/python-gsync/libgsync/sync/__init__.py",
  line 292, in _sync
    dst_file = self.dst.get_info(rel_path)
  File "/home/vic/github/python-gsync/libgsync/sync/file/remote.py",
  line 85, in get_info
    info = SyncFileInfo(**info)
  File "/home/vic/github/python-gsync/libgsync/sync/file/__init__.py",
  line 166, in __init__
    self.set_stat_info(description)
  File "/home/vic/github/python-gsync/libgsync/sync/file/__init__.py",
  line 262, in set_stat_info
    pickle.loads(content)

Error: ValueError('unsupported pickle protocol: 4',)
sent 0 bytes  received 0 bytes  0.00 bytes/sec
```

There is a problem with the protocol that `pickle` is using. It seems that `pickle` uses the protocol 2 when running `gsync` using Python 2. If I run `gsync` again with Python 3, some metadata is pickled again but with the default Python 3 protocol (3 or higher). If I try to use `gsync` again with Python 2, `pickle` does not know what to do with a protocol that is not supported.

**Comment by molinav** [comment]
The problem with `pickle` was solved in #11, so I close this.
